soul-admin/src/test/java/org/dromara/soul/admin/listener/websocket/WebsocketCollectorTest.java

```diff
@@ -0,0 +1,150 @@
+            @Override
+            public void onOpen(final ServerHandshake serverHandshake) {
+                log.info("Open connection");
+            }
+
+            @Override
+            public void onMessage(final String s) {
+                log.info("message : {}", s);
+            }
+
+            @Override
+            public void onClose(final int i, final String s, final boolean b) {
+                log.info("connection closed");
+            }
+
+            @Override
+            public void onError(final Exception e) {
+            }
+        };
+    }
+
+    @Test
+    public void testOnOpen() throws Exception {
+        client.connectBlocking();
+        Thread.sleep(1000);
+        assertEquals(1L, getSessionSetSize());
+        client.closeBlocking();
+    }
+
+    @Test
+    public void testOnMessage() throws Exception {
+        client.connectBlocking();
+        Thread.sleep(1000);
+        client.send(DataEventTypeEnum.MYSELF.name());
+        Thread.sleep(1000);
+        assertNotNull(getSession());
+        client.closeBlocking();
+    }
+
+    @Test
+    public void testOnClose() throws Exception {
+        client.connectBlocking();
+        Thread.sleep(1000);
+        assertEquals(1L, getSessionSetSize());
+        client.closeBlocking();
+        Thread.sleep(1000);
+        assertEquals(0L, getSessionSetSize());
+        assertNull(getSession());
+    }
+
+    @Test
+    public void testSend() throws Exception {
+        client.connectBlocking();
+        Thread.sleep(1000);
+        client.send(DataEventTypeEnum.MYSELF.name());
+        Thread.sleep(1000);
+        assertNotNull(getSession());
+        WebsocketCollector.send(null, DataEventTypeEnum.MYSELF);
+        WebsocketCollector.send("test", DataEventTypeEnum.MYSELF);
+        WebsocketCollector.send("test", DataEventTypeEnum.CREATE);
+        Thread.sleep(1000);
+        client.closeBlocking();
+    }
+
+    private long getSessionSetSize() throws ClassNotFoundException {
+        return ((Set) getField("SESSION_SET")).size();
+    }
+
+    private Session getSession() throws ClassNotFoundException {
+        return (Session) getField("session");
+    }
+
+    private Object getField(final String fieldName) throws ClassNotFoundException {
+        Class clazz = Class.forName("org.dromara.soul.admin.listener.websocket.WebsocketCollector");
+        Field[] declaredFields = clazz.getDeclaredFields();
+        return Arrays.stream(declaredFields)
+                .filter(each -> {
+                    each.setAccessible(true);
+                    return fieldName.equals(each.getName());
+                })
+                .findFirst()
+                .map(each -> {
+                    try {
+                        return each.get(clazz);
+                    } catch (IllegalAccessException e) {
+                        log.error("get field error", e);
+                        return null;
+                    }
+                }).orElse(null);
+    }
```

Comment:
Use `org.dromara.soul.common.utils.ReflectUtils` or `org.springframework.test.util.ReflectionTestUtils`

Reply:
ok, I will change the code.